.go-version

```diff
@@ -1 +1 @@
-1.23.4
+1.24.1
```

Comment:
I see that it was already updated in a previous PR (https://github.com/elastic/package-registry/pull/1285) the `toolchain` directive in `go.mod` :+1: 
https://github.com/elastic/package-registry/blob/07ca1502d2d426c844f4f8519890ad7f4f5d9710/go.mod#L5